File: Golang/768.最多能完成排序的块 II.go
package leetcode

import "sort"

func maxChunksToSorted(arr []int) int {
	ans := 0
	sum1, sum2 := 0, 0
	t := append([]int{}, arr...)
	sort.Slice(t, func(i, j int) bool { return t[i] < t[j] })
	for i := 0; i < len(arr); i++ {
		sum1 += arr[i]
		sum2 += t[i]
		if sum1 == sum2 {
			ans += 1
		}
	}
	return ans
}

File: TypeScript/768.最多能完成排序的块 II.ts
function maxChunksToSorted(arr: number[]): number {
    let ans: number = 0
    let [sum1, sum2]: number[] = [0, 0]
    const t: number[] = [...arr]
    t.sort((x, y) => x - y)
    for (let i = 0; i < arr.length; ++i) {
        sum1 += arr[i]
        sum2 += t[i]
        if (sum1 === sum2)
            ans += 1
    }
    return ans
};
